taichi/program/graph_builder.cpp

```diff
@@ -37,6 +37,13 @@ GraphBuilder::GraphBuilder() {
 
 Node *GraphBuilder::new_dispatch_node(Kernel *kernel,
                                       const std::vector<aot::Arg> &args) {
+  for (const auto &arg : args) {
+    if (all_args_.find(arg.name) != all_args_.end()) {
+      TI_ASSERT(all_args_[arg.name] == arg);
```

Comment:
nit: the user might get confused about the error message. Use a TI_ERROR instead?